src/workerd/io/worker-entrypoint.c++

```diff
@@ -402,6 +402,12 @@ kj::Promise<WorkerInterface::CustomEvent::Result>
   auto incomingRequest = kj::mv(KJ_REQUIRE_NONNULL(this->incomingRequest,
                                 "customEvent() can only be called once"));
   this->incomingRequest = nullptr;
+
+  auto& context = incomingRequest->getContext();
+  KJ_IF_MAYBE(t, incomingRequest->getWorkerTracer()) {
+      t->setEventInfo(context.now(), Trace::CustomEventInfo());
```

Comment:
I think this code belongs inside the custom event callback, not here. That way the event can potentially decide what type of info to add. Additionally, you would have avoided the need for #177 as you could have this code execute after `incomingRequest->delivered()`.

Reply:
Makes sense. I'll look into it tomorrow.